src/main/java/com/pretz/dyntreeserver/service/UserService.java

```diff
@@ -0,0 +1,69 @@
+package com.pretz.dyntreeserver.service;
+
+import com.pretz.dyntreeserver.domain.User;
+import com.pretz.dyntreeserver.repository.UserRepo;
+import com.pretz.dyntreeserver.service.exceptions.IncorrectPasswordException;
+import com.pretz.dyntreeserver.service.exceptions.UserCreationFailException;
+import com.pretz.dyntreeserver.service.exceptions.UserNotFoundException;
+import io.jsonwebtoken.Jwts;
+import io.jsonwebtoken.SignatureAlgorithm;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.security.crypto.password.PasswordEncoder;
+import org.springframework.stereotype.Service;
+
+import java.util.Date;
+
+import static com.pretz.dyntreeserver.service.Seed.SEED;
+
+@Service
+public class UserService {
+
+    private static final long TOKEN_VALID_TIME = 3600 * 1000; //1 hour in ms
+
+    private UserRepo userRepo;
+    private PasswordEncoder passwordEncoder;
+
+    @Autowired
+    public UserService(UserRepo userRepo, PasswordEncoder passwordEncoder) {
+        this.userRepo = userRepo;
+        this.passwordEncoder = passwordEncoder;
+    }
+
+    public String validateUser(User user) {
+        if (userRepo.findByName(user.getName()).isPresent()) {
```

Comment:
I'd propose a flat structure instead of if nesting, e.g.:
```
if (!userRepo[...].isPresent()) {
  throw new UserNotFoundException(user);
}

if (!passwordEncoder.matches([...])) {
  throw new IncorrectPasswordException(user);
}

return createToken(user);
```

that way you won't nest all the logic somewhere deep in the ifs and just have a simple flow: 
```
{
  // validation stuff
  [...]
  // proper logic
  [...]
}
```

much cleaner to read IMO

Reply:
Changed